agent/requestor/src/market.rs

```diff
@@ -69,6 +69,7 @@ pub(crate) async fn spawn_negotiations(
     allocation_size: i64,
     agreement_allocation: Arc<Mutex<HashMap<String, String>>>,
     agreement_tx: mpsc::Sender<String>,
+    one_agreement: bool,
```

Comment:
I think it is not necessary to inform this part of code that we want to process only one agreement
It is better to break the while loop which is processing agreements